romancal/lib/suffix.py

```diff
@@ -0,0 +1,196 @@
+"""Suffix manipulation
+
+API
+---
+`remove_suffix`
+    Main function used to remove suffixes from
+    file base names.
+
+Notes
+-----
+
+`KNOW_SUFFIXES` is the list used by `remove_suffix`. This
+list is generated by the function `combine_suffixes`. The function uses
+`SUFFIXES_TO_ADD` to add other suffixes that it would otherwise not
+discover or should absolutely be in the list. The function uses
+'SUFFIXES_TO_DISCARD` for strings found that are not to be considered
+suffixes.
+
+Hence, to update `KNOW_SUFFIXES`, update both `SUFFIXES_TO_ADD` and
+`SUFFIXES_TO_DISCARD` as necessary, then use the output of
+`find_suffixes`.
+"""
+import itertools
+import logging
+import re
+
+__all__ = ['remove_suffix']
+
+# Configure logging
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+
+# Suffixes that are hard-coded or otherwise
+# have to exist. Used by `find_suffixes` to
+# add to the result it produces.
+SUFFIXES_TO_ADD = [
+    'cal', 'crf',
```

Comment:
nitpicky, but break all of these up into individual lines if most of them are (unless theres a reason some of them are on the same line?)

Reply:
JWST has it in alphabetical order with each line listing a letter so there would only be 26 lines.  I kept that so as we add steps we can preserve that structure. We don't have to keep that.

Reply:
This if from JWST where the suffix list is broken up alphabetically with one line per letter. I think this will be useful as we add additional steps.